### ISSUE TITLE
no-unused-vars: Allow variables with `@export` to be unused

### ISSUE THREAD

**Issue by jschaf** [eslint/eslint#7302]
* The version of ESLint you are using.
v 3.3.1

* The problem you want to solve.
The [closure annotation `@export`](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler) isn't recognized as marking a variable as exported.

* Your take on the correct solution to problem.

If a variable has `@export` in an immediately preceding JSDOC comment, mark the variable as used.  The `@export` annotation should not need to restate the variable name.

```js
/** @export {number} */
var foo = 2;
```

Output
```
  2:5  warning  'foo' is defined but never used  no-unused-vars
```

**What actually happened? Please include the actual, raw output from ESLint.**

**Comment by nzakas** [comment]
We generally don't make changes to rules based on specific tools (otherwise ESLint would have to know about all tools anyone might be using).

We do have our [own syntax](http://eslint.org/docs/rules/no-unused-vars#exported) for marking variables as exported:

```js
// exported foo
```

I'd suggest putting that in your code.

**Comment by jschaf** [comment]
If I write a custom rule that marked variables with an `@export` using `markVariableAsUsed`, would that also work?

**Comment by platinumazure** [comment]
@jschaf You are definitely welcome to write a custom rule for this scenario, and I think it could work. Please feel free to stop by our [chat room](https://gitter.im/eslint/eslint) if you need help!
